### PR TITLE
api: catch UnexpectedResourceTypeState properly

### PR DESCRIPTION
**Pull request by jd**
When creating resource, this might happen if the resource type is not
completely created yet (race condition). Rather than crash and return 500,
return a 503 with a parseable reply so it can be retried.

### CODE REVIEW

File: gnocchi/rest/api.py [hunk 1]
```diff
@@ -1135,6 +1135,8 @@ def post(self):
             abort(400, six.text_type(e))
         except indexer.ResourceAlreadyExists as e:
             abort(409, six.text_type(e))
+        except indexer.UnexpectedResourceTypeState as e:
```
Comment (sileht) [inline gnocchi/rest/api.py]: In case of e.state is "creating" we can retry create resource or maybe raise 409 also

Comment (jd) [reply gnocchi/rest/api.py]: Do we want to do that server side? I'm not sure. Or maybe I'm lazy.

Comment (chungg) [reply gnocchi/rest/api.py]: i'm hesitant about 503. when i see 503, it's generally that service is completely offline/unaccessible.

Comment (jd) [reply gnocchi/rest/api.py]: Right. Maybe 500 would be better.

**Comment by jd** [comment]
Use 500 instead of 503

File: gnocchi/rest/api.py (hunk 1 above)
Comment (chungg) [reply gnocchi/rest/api.py]: i think i agree with sileht, 5xx anything is very aggressive to me.

Comment (sileht) [reply gnocchi/rest/api.py]: I don't see the point of catching a 500 for an other 500.

Comment (jd) [reply gnocchi/rest/api.py]: This one has a proper exception info returned?

**Pull request closed without merging**